src/main/java/com/twilio/oai/api/APIResourceBuilder.java

```diff
@@ -0,0 +1,205 @@
+package com.twilio.oai.api;
+
+import com.twilio.oai.DirectoryStructureService;
+import com.twilio.oai.HttpMethod;
+import com.twilio.oai.PathUtils;
+import com.twilio.oai.StringHelper;
+import com.twilio.oai.common.EnumConstants;
+import com.twilio.oai.resolver.IResolver;
+import com.twilio.oai.resolver.php.PhpParameterResolver;
+import com.twilio.oai.resolver.php.PhpPropertyResolver;
+import com.twilio.oai.template.ITemplate;
+import com.twilio.oai.template.PHPAPITemplate;
+import org.openapitools.codegen.CodegenModel;
+import org.openapitools.codegen.CodegenOperation;
+import org.openapitools.codegen.CodegenParameter;
+import org.openapitools.codegen.CodegenProperty;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.twilio.oai.common.ApplicationConstants.PATH_SEPARATOR_PLACEHOLDER;
+
+public class APIResourceBuilder implements IAPIResourceBuilder {
```

Comment:
A good deal of the code in here is duplicate from Node. If we're gong to add these abstractions, it would be best to refactor Node at the same time to not only show the benefit of such abstractions, but also ensure the interface supports multiple languages.